### ISSUE TITLE
Matrix math

### ISSUE THREAD

**Issue by ericremoreynolds** [mathnet/mathnet-symbolics#1]
Hello Christoph,

From your source code it seems to me that there is no support for matrix algebra and calculus in the library, is this correct?

Introducing matrix algebra is a pain, I have found, because all of a sudden some product expressions are commutative and others aren't.

Regards,

Eric.

**Comment by cdrnet** [comment]
Indeed, there is no support for matrix expressions. There was in old Math.NET Classic, but not in Symbolics yet.

**Comment by MovGP0** [comment]
I would skip matrices completely and go for tensors (#14) instead. 

This would make sense since scalars, vectors, and matrices are just a special kind of tensors.

**Comment by MovGP0** [comment]
the lack for matrices and tensors is annoying. it is currently not possible to solve system of equations symbolically: 

```fsharp
let burger = symbol "burger"
let pommes = symbol "pommes"
let coke = symbol "coke"
let bill = symbol "bill"

let equations = [
        (2*coke) == 30.0
        (1*coke + 2*burger) == 20.0 // equals does not exist
        (1*burger + 4*pommes) == 9.0
        (burger + pommes * coke) == bill
    ]

linSolve equations bill // not working
```
the only way i could figure out was the numerical solution: 
```fsharp
let M = matrix [[3.0; 0.0; 0.0]
                [1.0; 2.0; 0.0]
                [0.0; 1.0; 4.0]]

let R = vector [30.0; 20.0; 9.0]

let result = M.Solve(R)

let coke = result.[0]
let burger = result.[1]
let pommes = result.[2]

burger + pommes * coke // 15
```

maybe we could port [Redberry](http://redberry.cc/) ([Sourcecode](https://bitbucket.org/redberry/)), since it has great support for tensors.

**Comment by ingted** [comment]
[Quick way / ugly way to achieve vector/matrix support](https://github.com/ingted/mathnet-symbolics)
